pytest_html/plugin.py

```diff
@@ -376,12 +376,12 @@ def generate_summary_item(self):
         if self.rerun is not None:
             outcomes.append(Outcome('rerun', self.rerun))
 
-        summary = [html.h2('Summary'), html.p(
-            '{0} tests ran in {1:.2f} seconds. '.format(
-                numtests, suite_time_delta)),
-            html.p('(Un)check the boxes to filter the results.',
-                   class_='filter',
-                   hidden='true')]
+        summary = [html.h2('Summary'),
+                   self._generate_custom_summary(session.config),
+                   html.p('{0} tests ran in {1:.2f} seconds. '.format(
+                       numtests, suite_time_delta)),
+                   html.p('(Un)check the boxes to filter the results.',
+                          class_='filter', hidden='true')]
```

Comment:
It would be more powerful to call a `pytest_html_results_summary` with the complete summary list. The user can then decide to manipulate the list however they like. Alternatively, a `pytest_html_results_summary_prefix` or `pytest_html_results_summary_suffix` would preserve the default summary.

Reply:
Thinking about this further, we could have `pytest_html_results_summary` with three arguments: prefix, summary, and suffix. That would make it easy for the default summary to remain untouched, whilst allowing a prefix and suffix. For users that really want to, they can even manipulate the default summary.

Reply:
To make sure I understand you correctly:
- the `pytest_html_results_summary` hook will take three arguments: prefix, summary, suffix
- the argument type is a list of py.xml html objects
- the prefix by default contains the count/run time of the tests and the text about the filters (not sure about this one, alternative: default empty, always have count/run time and text)
- the summary by default contains the summary table
- the suffix by default is empty

Reply:
@davehunt Please ignore previous comment. I implemented the hook with three arguments:
- `prefix`: empty by default
- `summary`: contains the count/run time of the tests and the text about the filters 
- `prefix`: empty by default

Type of all three arguments is a list of py.xml html objects.